### PR TITLE
Remove unused authorization portal property

### PR DESCRIPTION
**Pull request by ao508**
Signed-off-by: Angelica Ochoa <aochoa4230@gmail.com>

### CODE REVIEW

File: core/src/main/java/org/mskcc/cbio/portal/util/internal/AccessControlImpl.java [hunk 1]
```diff
@@ -84,7 +84,7 @@
 
         //  Then, insert "All" Cancer Types at beginning  //TODO - fix this! It conflicts with ALL (Acute Lymphoid Leukemia)!
         ArrayList<CancerStudy> finalCancerStudiesList = new ArrayList<CancerStudy>();
-        String allCancerStudyTitle = (GlobalProperties.usersMustBeAuthorized()) ?
+        String allCancerStudyTitle = (GlobalProperties.usersMustAuthenticate()) ?
```
Comment (sheridancbio) [inline core/src/main/java/org/mskcc/cbio/portal/util/internal/AccessControlImpl.java:87]: Oh .. didn't realize we had this method available .. nice find.
I hope this covers the right cases through ... did you check whether this returns false for social_auth ?

File: docs/Authenticating-Users-via-LDAP.md [hunk 2]
```diff
@@ -13,8 +13,6 @@ In this setup LDAP will be used as a simple Identity Provider (IDP) for the auth
 
 * **authentication**:  a means of verifying that a user is who they purport to be.  Authentication is performed by the identify provider, by extracting the user name and password provided in a login request, and matching this with information stored in its repository.
 
-* **authorization**:  defines resources a user can access.  When authorization is turned *on* with cBioPortal, users can only access cancer studies they are specifically authorized to view.  This enables one to store multiple cancer studies within a single instance of cBioPortal, but provide fine-grained control over which users can access which studies.  Authorization is implemented within the core cBioPortal code, and *not* the identify provider.
```
Comment (sheridancbio) [inline docs/Authenticating-Users-via-LDAP.md:16]: I think maybe some of the info about authorities here could move into the authentication section above ... maybe a simple statement like "Once authenticated, users will have access to the cancer studies that they are authorized to view." And maybe we could add a link to the section of the documentation (if it exists) which explains setting user authorities.

**Review by sheridancbio — COMMENTED** [review]
will complete review soon

File: core/src/main/java/org/mskcc/cbio/portal/util/internal/AccessControlImpl.java (hunk 1 above)
Comment (ao508) [reply core/src/main/java/org/mskcc/cbio/portal/util/internal/AccessControlImpl.java:87]: @sheridancbio yep, it does 
```    public static boolean usersMustAuthenticate()
    {
        // authentication for social_auth is optional
        return (!authenticate.isEmpty() && !authenticate.equals("false") && !authenticate.equals("social_auth"));
    }
```

File: docs/Authenticating-Users-via-LDAP.md [hunk 3]
```diff
@@ -48,11 +46,7 @@ Please note that you will have to modify all the above to match your own setting
 
 ## Authorizing Users
 
-Within portal.properties, modify
-
-    authorization=true
-
-Next, please read the Wiki page on [User Authorization](User-Authorization.md), and add user rights for a single user.
+Please read the Wiki page on [User Authorization](User-Authorization.md), and add user rights for a single user.
```
Comment (sheridancbio) [inline docs/Authenticating-Users-via-LDAP.md:49]: Here maybe we should add a note that user authorizations only apply to portal deployments which require Authentication, and are not necessary for portals that use authentication=false or authentication=social_auth

File: docs/Authenticating-Users-via-SAML.md [hunk 4]
```diff
@@ -18,8 +18,6 @@ In its simplest terms, SAML boils down to four terms:
 
 * **authentication**:  a means of verifying that a user is who they purport to be.  Authentication is performed by the identify provider, by extracting the user name and password provided in a login form, and matching this with information stored in a database.
 
-* **authorization**:  defines resources a user can access.  When authorization is turned on with cBioPortal, users can only access cancer studies they are specifically authorized to view (or PUBLIC studies, [if the server has been configured to contain these](User-Authorization.md#configuring-public-studies)).  This enables one to store multiple cancer studies within a single instance of cBioPortal, but provide fine-grained control over which users can access which studies.  Authorization is implemented within the core cBioPortal code, and not the identify provider.
```
Comment (sheridancbio) [inline docs/Authenticating-Users-via-SAML.md:21]: similar comment as above. Here they mention "public studies" which is no longer a functional feature if I understood @n1zea144 correctly .. so it is good we are removing this reference.

File: docs/Authenticating-Users-via-SAML.md [hunk 5]
```diff
@@ -134,8 +132,6 @@ Within portal.properties, make sure that:
 
 Then, modify the section labeled `authentication`. See SAML parameters shown in example below:
 
-    # authentication
```
Comment (sheridancbio) [inline docs/Authenticating-Users-via-SAML.md:137]: Doesn't this section a line like authentication=saml ?
I'm not sure, but I'm surprised it is not here. It seems like the authentication property is missing from all these sections, so maybe I missed something or am confused.

Ah, I see a reference to setting the method on the command line like this CATALINA_OPTS='-Dauthenticate=saml' (in portal.properties-Reference.md). I think a quick test of the proper authorization when setting the authentication through CATALINA_OPTS under both googleplus and social_auth would be a good idea.

So maybe the documentation now use this as the primary way to set the authentication method.

**Review by sheridancbio — APPROVED** [review]
some suggestions included, but looks good

**Pull request merged**